pkg/client/event/godoc.go

```diff
@@ -0,0 +1,12 @@
+// Package event provides a generic client for subscribing to on-chain events
```

Comment:
I feel like it would make more sense for this and `pkg/client/events_query` **to be the same**. However, upon reflection, I don't see any reason not to simplify the name to `pkg/client/events`, like you have here.